### PR TITLE
[Dubbo-900] Fix 通过 override 修改 hessian协议的提供者的配置 不生效 #900

### PR DESCRIPTION
**Pull request by CrazyHZM**
## What is the purpose of the change
When modifying the configuration through override, you need to re-expose the newly modified service.
Compare the new url with the old url. If they are the same, they will be directly fetched from the collection. If they are different, re-create an export.

fix https://github.com/apache/incubator-dubbo/issues/900

## Brief changelog

XXXXX

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/protocol/AbstractProxyProtocol.java [hunk 1]
```diff
@@ -67,7 +67,11 @@ public void setProxyFactory(ProxyFactory proxyFactory) {
         Exporter<T> exporter = (Exporter<T>) exporterMap.get(uri);
         if (exporter != null) {
             // When modifying the configuration through override, you need to re-expose the newly modified service.
-            if (exporter.getInvoker().getUrl().equals(invoker.getUrl())) {
+            if ("hessian".equals(invoker.getUrl().getProtocol())) {
+                if (exporter.getInvoker().getUrl().equals(invoker.getUrl())) {
```
Comment (satansk) [inline dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/protocol/AbstractProxyProtocol.java]: Could `getUrl` return null here? If so, `Objects.equals` might be better.

Comment (CrazyHZM) [reply dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/protocol/AbstractProxyProtocol.java]: You are right, I will modify it.

File: dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/protocol/AbstractProxyProtocol.java [hunk 2]
```diff
@@ -66,7 +67,14 @@ public void setProxyFactory(ProxyFactory proxyFactory) {
         final String uri = serviceKey(invoker.getUrl());
         Exporter<T> exporter = (Exporter<T>) exporterMap.get(uri);
         if (exporter != null) {
-            return exporter;
+            // When modifying the configuration through override, you need to re-expose the newly modified service.
+            if ("hessian".equals(invoker.getUrl().getProtocol())) {
```
Comment (beiwei30) [inline dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/protocol/AbstractProxyProtocol.java]: We should not add special check for "hessian". It looks like we should compare URL for all proxy protocol when the cache is hit.

**Comment by CrazyHZM** [comment]
> pls. check my comments, we should not add 'hessian' special check here.

I have modified it.

**Pull request merged**